src/charm.py

```diff
@@ -114,15 +126,38 @@ def _on_config_changed(self, event):
         self.unit.status = WaitingStatus("configuring temporal")
         self._update(event)
 
+    @log_event_handler(logger)
+    def _on_restart(self, event):
+        """Restart Temporal ui action handler.
+
+        Args:
+            event:The event triggered by the restart action
+        """
+        container = self.unit.get_container(self.name)
+        if not container.can_connect():
+            event.defer()
+            return
+
+        self.unit.status = MaintenanceStatus("restarting ui")
```

Comment:
Will this just remain in MaintenanceStatus after restart? You could set an active status after restart, or set up some [checks](https://juju.is/docs/sdk/interact-with-pebble#heading--fetch-check-status) to validate the app has restarted

Reply:
I am also interested in this.

Reply:
You're right, i've updated a check to set the active status after restarting. Thanks!